File: app/src/main/java/com/gaurav/newsapp/viewmodel/NewsViewModelFactory.java
package com.gaurav.newsapp.viewmodel;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import com.gaurav.newsapp.data.NewsDataRepository;

public class NewsViewModelFactory implements ViewModelProvider.Factory {

    private NewsDataRepository repository;

    public NewsViewModelFactory(NewsDataRepository repository){

        this.repository = repository;
    }
    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new NewsViewModel(repository);
    }
}

File: app/src/main/java/com/gaurav/newsapp/viewmodel/NewsViewModelFactory.kt
package com.gaurav.newsapp.viewmodel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.gaurav.newsapp.data.NewsDataRepository

class NewsViewModelFactory(private val repository: NewsDataRepository) : ViewModelProvider.Factory {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return NewsViewModel(repository) as T
    }

}